### PR TITLE
added trongrid JWT feature

### PR DESCRIPTION
**Pull request by jake-ephraim**
Aside the api key, TronGrid supports JWT with the RS256 algorithm.

### CODE REVIEW

File: tronpy/providers/http.py [hunk 1]
```diff
@@ -56,6 +58,11 @@ def __init__(
             self._default_api_keys = self._api_keys.copy()
         else:
             self.use_api_key = False
+        
+        if jw_token is None or isinstance(jw_token, str):
```
Comment (MrNaif2018) [inline tronpy/providers/http.py:62]: For case of `js_token` being None, it shouldn't set the attribute to `"Bearer "`, but instead not add it to the headers at all

File: tronpy/providers/async_http.py [hunk 2]
```diff
@@ -22,6 +23,7 @@ def __init__(
         timeout: float = DEFAULT_TIMEOUT,
         client: httpx.AsyncClient = None,
         api_key: str = DEFAULT_API_KEY,
+        jw_token: str = None,
```
Comment (MrNaif2018) [inline tronpy/providers/async_http.py:26]: Doesn't async client need same kind of validation like in sync one? Also check the comment below

File: tronpy/providers/http.py [hunk 3]
```diff
@@ -56,6 +58,11 @@ def __init__(
...
+            self.jw_token = f"Bearer {jw_token}"
+        else:
+            raise TypeError("unknown jwt type {}".format(jw_token))
```
Comment (MrNaif2018) [inline tronpy/providers/http.py:65]: In general I think though that something like

self.jw_token = jw_token

...

if self.jw_token:
     self.sess.headers["Authorization"] = f"Bearer {jw_token}"

**Review by MrNaif2018 — CHANGES_REQUESTED** [review]
Hi! Thank you for your PR! Please check out the review comments

File: tronpy/providers/http.py (hunk 1 above)
Comment (jake-ephraim) [reply tronpy/providers/http.py:62]: okay, thanks.

File: tronpy/providers/http.py (hunk 3 above)
Comment (jake-ephraim) [reply tronpy/providers/http.py:65]: i was thinking to follow the existing code format by adding the headers in the `make_request` method

File: tronpy/providers/async_http.py (hunk 2 above)
Comment (jake-ephraim) [reply tronpy/providers/async_http.py:26]: okay, i will include it.

**Comment by jake-ephraim** [comment]
i have made the changes.

**Comment by jake-ephraim** [comment]
I think it's good now, right?

**Review by MrNaif2018 — APPROVED** [review]
I've fixed the remaining issues. Thank you for your PR!

**Pull request merged**